### PR TITLE
ref: Replace chrono usage with time/SystemTime

### LINKED ISSUE #375: RUSTSEC-2020-0159: Potential segfault in `localtime_r` invocations
> Potential segfault in `localtime_r` invocations

| Details             |                                                |
| ------------------- | ---------------------------------------------- |
| Package             | `chrono`                      |
| Version             | `0.4.19`                   |
| URL                 | [https://github.com/chronotope/chrono/issues/499](https://github.com/chronotope/chrono/issues/499) |
| Date                | 2020-11-10                         |

### Impact

Unix-like operating systems may segfault due to dereferencing a dangling pointer in specific circumstances. This requires an environment variable to be set in a different thread than the affected functions. This may occur without the user&#39;s knowledge, notably in a third-party library.

### Workarounds

No workarounds are known.

### References

- [time-rs/time#293](https://github.com/time-rs/time/issues/293)

See [advisory page](https://rustsec.org/advisories/RUSTSEC-2020-0159.html) for additional details.

### PR DESCRIPTION
**Pull request by Swatinem**
The public API only uses `SystemTime`, internally we use `time:0.3` for formatting and parsing.

fixes #375

### CODE REVIEW

File: sentry-core/src/session.rs [hunk 1]
```diff
@@ -257,9 +255,13 @@ impl SessionFlusher {
             attributes: session_update.attributes.clone(),
         });
 
-        let started = session_update
+        let duration = session_update
             .started
-            .duration_trunc(ChronoDuration::minutes(1))
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap();
+        let duration = (duration.as_secs() / 60) * 60;
```
Comment (loewenheim) [inline sentry-core/src/session.rs:262]: What is this intended to do?

Comment (Swatinem) [reply sentry-core/src/session.rs:262]: Integer division is truncating, so this should round down to the nearest minute.

Comment (loewenheim) [reply sentry-core/src/session.rs:262]: Yeah I figured, I just don't really understand why we're rounding it when we output seconds anyway.

Comment (Swatinem) [reply sentry-core/src/session.rs:262]: When we pre-aggregate sessions on the client, we group them into minute buckets based on the truncated systemtime ;-)

Comment (loewenheim) [reply sentry-core/src/session.rs:262]: I see, thanks for the explanation :)

**Review by loewenheim — APPROVED** [review]

**Comment by Swatinem** [comment]
Oh well: 

```
error: failed to parse manifest at `/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.3.5/Cargo.toml`

Caused by:
  feature `resolver` is required

  this Cargo does not support nightly features, but if you
  switch to nightly channel you can add
  `cargo-features = ["resolver"]` to enable this feature
```

Guess we gotta bump our MSRV maybe?

https://github.com/time-rs/time/commit/e4d09e20e5c44d20ae4c68aa0797c672dd0e713d

**Pull request merged**